client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,13 @@
+import { Route, RouteProps, Redirect } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+
+const ProtectedRoute = ({ ...routeProps }: RouteProps) => {
+  const { loggedInUser } = useAuth();
+
```

Comment:
Optional: Additionally, we can also display a loading spinner while we fetch the logged in user's details (for example, on page refresh we will be directed to the login page for a split second)